wicket-core/src/main/java/org/apache/wicket/markup/html/form/upload/OnFilesSelectedBehavior.java

```diff
@@ -0,0 +1,134 @@
+ * s	 *
+ *
+ * @author Ernesto Reinaldo Barreiro (reiern70@gmail.com).
+ */
+public abstract class OnFilesSelectedBehavior extends OnChangeAjaxBehavior {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final ResourceReference JS = new JavaScriptResourceReference(OnFilesSelectedBehavior.class, "OnFilesSelectedBehavior.js");
+
+    @Override
+    protected void onBind() {
+        super.onBind();
+        Component component = getComponent();
+        if (!(component instanceof FileUploadField)) {
+            throw new WicketRuntimeException("Behavior " + getClass().getName()
+                    + " can only be added to an instance of a FileUploadField");
+        }
+    }
+
+    @Override
+    protected void onUpdate(AjaxRequestTarget target) {
+        Request request = RequestCycle.get().getRequest();
+        List<FileDescription> fileDescriptions = new ArrayList<>();
+        IRequestParameters parameters = request.getRequestParameters();
+        // data is streamed as JSON.
+        StringValue fileInfos = parameters.getParameterValue("fileInfos");
+        JSONArray jsonArray = new JSONArray(fileInfos.toString());
+        for (int i = 0; i < jsonArray.length(); i++) {
+            fileDescriptions.add(new FileDescription((JSONObject)jsonArray.get(i)));
+        }
+        onSelected(target, fileDescriptions);
+    }
+
+
+    /**
+     * Called when a file, at client side is selected.
+     *
+     * @param target           The {@link org.apache.wicket.ajax.AjaxRequestTarget}
+     * @param fileDescriptions A list of FileDescription
+     */
+    protected abstract void onSelected(AjaxRequestTarget target, List<FileDescription> fileDescriptions);
+
+    @Override
+    protected void updateAjaxAttributes(AjaxRequestAttributes attributes)
+    {
+        super.updateAjaxAttributes(attributes);
+        attributes.getAjaxCallListeners().add(new IAjaxCallListener()
+        {
+            @Override
+            public CharSequence getPrecondition(Component component)
+            {
+                return "return Wicket.OnFilesSelected.precondition(this, '" + component.getMarkupId() + "');";
+            }
+        });
+        attributes.getDynamicExtraParameters().add("return Wicket.OnFilesSelected.collectFilesDetails('" + getComponent().getMarkupId() + "');");
+    }
+
+    @Override
+    public void renderHead(Component component, IHeaderResponse response)
+    {
+        response.render(JavaScriptHeaderItem.forReference(JS));
+        super.renderHead(component, response);
+    }
+
+    /**
+     * Creates an {@link OnFilesSelectedBehavior} based on lambda expressions
+     *
+     * @param select {@link SerializableBiConsumer}
+     *
+     * @return the {@link org.apache.wicket.markup.html.form.upload.OnFilesSelectedBehavior} behavior
+     */
+    public static OnFilesSelectedBehavior select(
+            SerializableBiConsumer<AjaxRequestTarget, List<FileDescription>> select)
```

Comment:
For consistency with the other similar methods the method name should be `onSelect`
https://github.com/apache/wicket/blob/048e5681df960018722237ee6254273e839b5fd2/wicket-core/src/main/java/org/apache/wicket/ajax/AjaxEventBehavior.java#L166

Reply:
Hum... Maybe I understood not correctly what @svenmeier was asking?

Reply:
@svenmeier said "file.add(OnFileSelectedBehavior.onSelected"
I am fine with either `onSelect` and `onSelected` but I like `onSelect` better, it is more consistent with the rest similar methods.

Reply:
![image](https://user-images.githubusercontent.com/462655/114366373-64250e00-9b49-11eb-8a87-238ffff57d87.png)

?

Reply:
Here, I think, he meant the class name, not the method name,

Reply:
renamed behavior to FilesSelectedBehavior and factory method to  onSelected.